qir/qat/Validator/Tests/Unit/CirqTypeQirsPositive.cpp

```diff
@@ -0,0 +1,173 @@
+  %5 = inttoptr i64 5 to %Qubit*
+  %6 = inttoptr i64 6 to %Qubit*
+  %7 = inttoptr i64 7 to %Qubit*
+  call void @__quantum__qis__x__body(%Qubit* %2)
+  call void @__quantum__qis__x__body(%Qubit* %6)
+  call void @__quantum__qis__x__body(%Qubit* %5)
+  call void @__quantum__qis__x__body(%Qubit* %0)
+  call void @__quantum__qis__s__body(%Qubit* %2)
+  call void @__quantum__qis__s__body(%Qubit* %6)
+  call void @__quantum__qis__s__body(%Qubit* %5)
+  call void @__quantum__qis__s__body(%Qubit* %0)
+  call void @__quantum__qis__cx__body(%Qubit* %2, %Qubit* %4)
+  call void @__quantum__qis__h__body(%Qubit* %2)
+  call void @__quantum__qis__cx__body(%Qubit* %4, %Qubit* %2)
+  call void @__quantum__qis__rz__body(double 0x3FD190467F77F61E, %Qubit* %2)
+  call void @__quantum__qis__cx__body(%Qubit* %4, %Qubit* %2)
+  call void @__quantum__qis__rz__body(double 0xBFD190467F77F61E, %Qubit* %2)
+  call void @__quantum__qis__h__body(%Qubit* %2)
+  call void @__quantum__qis__cx__body(%Qubit* %2, %Qubit* %4)
+  call void @__quantum__qis__cx__body(%Qubit* %6, %Qubit* %7)
+  call void @__quantum__qis__h__body(%Qubit* %6)
+  call void @__quantum__qis__cx__body(%Qubit* %7, %Qubit* %6)
+  call void @__quantum__qis__rz__body(double 0x3FE069081E98A77B, %Qubit* %6)
+  call void @__quantum__qis__cx__body(%Qubit* %7, %Qubit* %6)
+  call void @__quantum__qis__rz__body(double 0xBFE069081E98A77B, %Qubit* %6)
+  call void @__quantum__qis__h__body(%Qubit* %6)
+  call void @__quantum__qis__cx__body(%Qubit* %6, %Qubit* %7)
+  call void @__quantum__qis__cx__body(%Qubit* %5, %Qubit* %3)
+  call void @__quantum__qis__s__body(%Qubit* %3)
+  call void @__quantum__qis__cx__body(%Qubit* %3, %Qubit* %5)
+  call void @__quantum__qis__cx__body(%Qubit* %5, %Qubit* %3)
+  call void @__quantum__qis__cx__body(%Qubit* %0, %Qubit* %1)
+  call void @__quantum__qis__h__body(%Qubit* %0)
+  call void @__quantum__qis__cx__body(%Qubit* %1, %Qubit* %0)
+  call void @__quantum__qis__rz__body(double 0x3FE069081E98A785, %Qubit* %0)
+  call void @__quantum__qis__cx__body(%Qubit* %1, %Qubit* %0)
+  call void @__quantum__qis__rz__body(double 0xBFE069081E98A785, %Qubit* %0)
+  call void @__quantum__qis__h__body(%Qubit* %0)
+  call void @__quantum__qis__cx__body(%Qubit* %0, %Qubit* %1)
+  call void @__quantum__qis__t__body(%Qubit* %2)
+  call void @__quantum__qis__s__body(%Qubit* %2)
+  call void @__quantum__qis__t__body(%Qubit* %6)
+  call void @__quantum__qis__s__body(%Qubit* %6)
+  call void @__quantum__qis__t__body(%Qubit* %5)
+  call void @__quantum__qis__s__body(%Qubit* %5)
+  call void @__quantum__qis__t__body(%Qubit* %0)
+  call void @__quantum__qis__s__body(%Qubit* %0)
+  call void @__quantum__qis__x__body(%Qubit* %2)
+  call void @__quantum__qis__x__body(%Qubit* %6)
+  call void @__quantum__qis__x__body(%Qubit* %5)
+  call void @__quantum__qis__x__body(%Qubit* %0)
+
+  )script");
+}
+
+TEST(CirqPositive, DISABLED_BernsteinVazirani)
+{
+    // m operations is having the wrong signature
```

Comment:
This test could be enabled if new variables `%9` through `%17` were declared with values like the qubits (ie: `inttoptr i64 0 to %Result*`) and the calls to measurement updated to the pattern `call void @__quantum__qis__mz__body(%Qubit* 0, %Result* 0)`. The declaration on like 36 would need to be updated too.

Reply:
This test is actually disabled for the time being. I have to fix my ad-hoc cirq converter or wait for the real one to arrive to get the code right here.

Reply:
I've removed this code.